src/http/file.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
What can make this a bit tricky is that File should should extend Blob.
See how I solved this for `Buffer` where it extends/inherits `Uint8Array`. So you can rely on the procedural macros have to fall back to adding the additional methods (name etc) to the prototype. Example here:
https://github.com/awslabs/llrt/blob/30117081660d638bbd9dce99b9da25251460e278/src/buffer.rs#L180C1-L200C2.

Reply:
Here is an example of inheritance when we have two rust defined "classes":
[rustexplorer.com](https://www.rustexplorer.com/b#LyoKW2RlcGVuZGVuY2llc10KcnF1aWNranMgPSB7IHZlcnNpb24gPSAiMC41LjEiLCBmZWF0dXJlcyA9IFsKICAgICJmdWxsLWFzeW5jIiwKICAgICJwYXJhbGxlbCIsCl0sIGRlZmF1bHQtZmVhdHVyZXMgPSBmYWxzZSB9CnRva2lvID0geyB2ZXJzaW9uID0gIjEiLCBmZWF0dXJlcyA9IFsiZnVsbCJdIH0KKi8KCnVzZSBycXVpY2tqczo6ewogICAgY2xhc3M6OkpzQ2xhc3MsCiAgICBsb2FkZXI6OntCdWlsdGluUmVzb2x2ZXIsIEZpbGVSZXNvbHZlciwgTW9kdWxlTG9hZGVyLCBTY3JpcHRMb2FkZXJ9LAogICAgcHJlbHVkZTo6e0Z1bmMsIFRoaXN9LAogICAgQXN5bmNDb250ZXh0LCBBc3luY1J1bnRpbWUsIENhdGNoUmVzdWx0RXh0LCBDbGFzcywgQ3R4LCBFeGNlcHRpb24sIEludG9KcywgTnVsbCwgT2JqZWN0LAogICAgUmVzdWx0LCBWYWx1ZSwKfTsKCmZuIHByaW50PCdqcz4oY3R4OiBDdHg8J2pzPiwgdmFsOiBWYWx1ZTwnanM+KSAtPiBSZXN1bHQ8KCk+IHsKICAgIGxldCBzdHIgPSBtYXRjaCBjdHguanNvbl9zdHJpbmdpZnlfcmVwbGFjZXJfc3BhY2UodmFsLCBOdWxsLCAiICAiLmludG9fanMoJmN0eCkpPyB7CiAgICAgICAgU29tZShzdHIpID0+IHN0ci50b19zdHJpbmcoKSwKICAgICAgICBOb25lID0+IE9rKCJ1bmRlZmluZWQiLmludG8oKSksCiAgICB9PzsKCiAgICBwcmludGxuISgie30iLCBzdHIpOwogICAgT2soKCkpCn0KCiNbcnF1aWNranM6OmNsYXNzXQojW2Rlcml2ZShycXVpY2tqczo6Y2xhc3M6OlRyYWNlKV0Kc3RydWN0IFN1YkNsYXNzIHsKICAgIHN1Yl92YWx1ZTogU3RyaW5nLAogICAgYmFzZV9jbGFzczogQmFzZUNsYXNzLAp9CgojW3JxdWlja2pzOjptZXRob2RzXQppbXBsIFN1YkNsYXNzIHsKICAgICNbcWpzKGNvbnN0cnVjdG9yKV0KICAgIGZuIG5ldzwnanM+KHN1Yl92YWx1ZTogU3RyaW5nLCBzdXBlcl92YWx1ZTogdXNpemUpIC0+IFNlbGYgewogICAgICAgIFNlbGYgewogICAgICAgICAgICBzdWJfdmFsdWUsCiAgICAgICAgICAgIGJhc2VfY2xhc3M6IEJhc2VDbGFzczo6bmV3KHN1cGVyX3ZhbHVlKSwKICAgICAgICB9CiAgICB9CgogICAgZm4gdmFsdWUyKCZzZWxmKSAtPiBTdHJpbmcgewogICAgICAgIHJldHVybiBzZWxmLnN1Yl92YWx1ZS5jbG9uZSgpOwogICAgfQp9CgojW3JxdWlja2pzOjpjbGFzc10KI1tkZXJpdmUocnF1aWNranM6OmNsYXNzOjpUcmFjZSldCnN0cnVjdCBCYXNlQ2xhc3MgewogICAgbnVtYmVyOiB1c2l6ZSwKfQoKI1tycXVpY2tqczo6bWV0aG9kc10KaW1wbCBCYXNlQ2xhc3MgewogICAgI1txanMoY29uc3RydWN0b3IpXQogICAgZm4gbmV3KG51bWJlcjogdXNpemUpIC0+IFNlbGYgewogICAgICAgIFNlbGYgeyBudW1iZXIgfQogICAgfQoKICAgIC8vYmFzZSBjbGFzc2VzIGhhdmUgdG8gdXNlIHRoaXM6IFRoaXM8T2JqZWN0PCdqcz4+CiAgICBmbiB2YWx1ZTwnanM+KGN0eDogQ3R4PCdqcz4sIHRoaXM6IFRoaXM8T2JqZWN0PCdqcz4+KSAtPiBSZXN1bHQ8dXNpemU+IHsKICAgICAgICBpZiBsZXQgU29tZSh0aGlzKSA9IENsYXNzOjo8J2pzLCBTZWxmPjo6ZnJvbV9vYmplY3QodGhpcy4wLmNsb25lKCkpIHsKICAgICAgICAgICAgcmV0dXJuIE9rKHRoaXMuYm9ycm93KCkubnVtYmVyKTsKICAgICAgICB9CgogICAgICAgIGlmIGxldCBTb21lKHRoaXMpID0gQ2xhc3M6OjwnanMsIFN1YkNsYXNzPjo6ZnJvbV9vYmplY3QodGhpcy4wKSB7CiAgICAgICAgICAgIHJldHVybiBPayh0aGlzLmJvcnJvdygpLmJhc2VfY2xhc3MubnVtYmVyKTsKICAgICAgICB9CgogICAgICAgIEVycihFeGNlcHRpb246OnRocm93X3JlZmVyZW5jZSgKICAgICAgICAgICAgJmN0eCwKICAgICAgICAgICAgJlsiQmFkIGFyZ3VtZW50LCAndGhpcycgaXMgbm90IGEgIiwgU2VsZjo6TkFNRV0uY29uY2F0KCksCiAgICAgICAgKSkKICAgIH0KfQoKI1t0b2tpbzo6bWFpbl0KYXN5bmMgZm4gbWFpbigpIC0+IGNvcmU6OnJlc3VsdDo6UmVzdWx0PCgpLCBCb3g8ZHluIHN0ZDo6ZXJyb3I6OkVycm9yPj4gewogICAgbGV0IHJlc29sdmVyID0gKAogICAgICAgIEJ1aWx0aW5SZXNvbHZlcjo6ZGVmYXVsdCgpLAogICAgICAgIEZpbGVSZXNvbHZlcjo6ZGVmYXVsdCgpLndpdGhfcGF0aCgiLiIpLAogICAgKTsKICAgIGxldCBsb2FkZXIgPSAoTW9kdWxlTG9hZGVyOjpkZWZhdWx0KCksIFNjcmlwdExvYWRlcjo6ZGVmYXVsdCgpKTsKCiAgICBsZXQgcnQgPSBBc3luY1J1bnRpbWU6Om5ldygpLnVud3JhcCgpOwogICAgcnQuc2V0X21heF9zdGFja19zaXplKDUxMiAqIDEwMjQpLmF3YWl0OwogICAgcnQuc2V0X2xvYWRlcihyZXNvbHZlciwgbG9hZGVyKS5hd2FpdDsKCiAgICBsZXQgY3R4OiBBc3luY0NvbnRleHQgPSBBc3luY0NvbnRleHQ6OmZ1bGwoJnJ0KS5hd2FpdC51bndyYXAoKTsKCiAgICBsZXQgcmVzID0gY3R4CiAgICAgICAgLndpdGgofGN0eHwgLT4gcnF1aWNranM6OlJlc3VsdDwoKT4gewogICAgICAgICAgICBsZXQgZ2xvYmFscyA9IGN0eC5nbG9iYWxzKCk7CgogICAgICAgICAgICBDbGFzczo6PEJhc2VDbGFzcz46OmRlZmluZSgmZ2xvYmFscyk/OwogICAgICAgICAgICBDbGFzczo6PFN1YkNsYXNzPjo6ZGVmaW5lKCZnbG9iYWxzKT87CgogICAgICAgICAgICAvL2NvbmZpZ3VyZSBpbmhlcml0YW5jZSBieSBzZXR0aW5nIHRoZSBfX3Byb3RvX18gcHJvcGVydHkgb2YgdGhlIHN1YiBjbGFzcyB0byB0aGUgYmFzZSBjbGFzcwogICAgICAgICAgICBsZXQgYmFzZV9wcm90byA9IENsYXNzOjo8QmFzZUNsYXNzPjo6cHJvdG90eXBlKGN0eC5jbG9uZSgpKTsKICAgICAgICAgICAgbGV0IHN1Yl9wcm90byA9IENsYXNzOjo8U3ViQ2xhc3M+Ojpwcm90b3R5cGUoY3R4LmNsb25lKCkpLnVud3JhcCgpOwogICAgICAgICAgICBzdWJfcHJvdG8uc2V0X3Byb3RvdHlwZShiYXNlX3Byb3RvLmFzX3JlZigpKT87CgogICAgICAgICAgICBsZXQgY29uc29sZSA9IE9iamVjdDo6bmV3KGN0eC5jbG9uZSgpKT87CiAgICAgICAgICAgIGNvbnNvbGUuc2V0KCJsb2ciLCBGdW5jOjpmcm9tKHByaW50KSk/OwogICAgICAgICAgICBnbG9iYWxzLnNldCgiY29uc29sZSIsIGNvbnNvbGUpPzsKCiAgICAgICAgICAgIC8vbGV0IGpzX3NvdXJjZSA9IGZzOjpyZWFkX3RvX3N0cmluZygiaW5kZXgubWpzIik/OwoKICAgICAgICAgICAgY3R4LmV2YWwoCiAgICAgICAgICAgICAgICByIyIKCmNvbnN0IGJhc2UgPSBuZXcgQmFzZUNsYXNzKDEpOwoKY29uc3QgZGVyaXZlID0gbmV3IFN1YkNsYXNzKCJoZWxsbyBzdWIiLCAxMDApOwpjb25zb2xlLmxvZyhkZXJpdmUudmFsdWUoKSk7CmNvbnNvbGUubG9nKGRlcml2ZS52YWx1ZTIoKSk7CmNvbnNvbGUubG9nKGRlcml2ZSBpbnN0YW5jZW9mIEJhc2VDbGFzcyk7CmNvbnNvbGUubG9nKGRlcml2ZS5fX3Byb3RvX18uX19wcm90b19fLnZhbHVlLmNhbGwoZGVyaXZlKSk7CmNvbnNvbGUubG9nKGJhc2UudmFsdWUoKSk7CiAgICAgICAgCiAgICAgICAgICAgICIjLAogICAgICAgICAgICApPzsKCiAgICAgICAgICAgIE9rKCgpKQogICAgICAgIH0pCiAgICAgICAgLmF3YWl0OwoKICAgIGN0eC53aXRoKHxjdHh8IHsKICAgICAgICBpZiBsZXQgRXJyKGVycikgPSByZXMuY2F0Y2goJmN0eCkgewogICAgICAgICAgICBlcHJpbnRsbiEoInt9IiwgZXJyKTsKICAgICAgICB9CiAgICB9KQogICAgLmF3YWl0OwogICAgcnQuaWRsZSgpLmF3YWl0OwoKICAgIGRyb3AoY3R4KTsKICAgIGRyb3AocnQpOwoKICAgIE9rOjo8XywgQm94PGR5biBzdGQ6OmVycm9yOjpFcnJvcj4+KCgpKQp9Cg==)